Model/Job.php

```diff
@@ -124,44 +134,17 @@ public function canMerge(Job $job, $maxJobDataSize)
 
         $decodedData = $this->getDecodedData();
 
-        // @todo Remove legacy checks on 3.16.0
-        if ((!isset($decodedData['product_ids']) || count($decodedData['product_ids']) <= 0)
-            && (!isset($decodedData['category_ids']) || count($decodedData['category_ids']) < 0)
-            && (!isset($decodedData['entity_ids']) || count($decodedData['entity_ids']) < 0)
-            && (!isset($decodedData['page_ids']) || count($decodedData['page_ids']) < 0)) {
+        if (!isset($decodedData['entityIds']) || count($decodedData['entityIds']) <= 0) {
             return false;
         }
 
         $candidateDecodedData = $job->getDecodedData();
 
-        // @todo Remove legacy checks on 3.16.0
-        if ((!isset($candidateDecodedData['product_ids']) || count($candidateDecodedData['product_ids']) <= 0)
-            && (!isset($candidateDecodedData['category_ids']) || count($candidateDecodedData['category_ids']) < 0)
-            && (!isset($candidateDecodedData['entity_ids']) || count($candidateDecodedData['entity_ids']) < 0)
-            && (!isset($candidateDecodedData['page_ids']) || count($candidateDecodedData['page_ids']) < 0)) {
-            return false;
-        }
-
-        // @todo Remove on 3.16.0
-        if (isset($decodedData['product_ids'])
-            && count($decodedData['product_ids']) + count($candidateDecodedData['product_ids']) > $maxJobDataSize) {
-            return false;
-        }
-
-        // @todo Remove on 3.16.0
-        if (isset($decodedData['category_ids'])
-            && count($decodedData['category_ids']) + count($candidateDecodedData['category_ids']) > $maxJobDataSize) {
+        if (!isset($candidateDecodedData['entityIds']) || count($candidateDecodedData['entityIds']) <= 0) {
```

Comment:
This looks like it may have been breaking the merging of newer style job records so it's good we're removing the old stuff. 
Was this also affecting 3.16 or only 3.17?